__brick__/web/flutter_bootstrap.js

```diff
@@ -31,48 +31,44 @@ async function beginPreloading() {
   progressIndicator.style.width = '0%';
   progressText.textContent = `Loaded ${loadedAssets} of ${totalAssets} assets`;
 
-  async function reportProgress() {
-    loadedAssets++;
-
-    const value = Math.floor((loadedAssets / totalAssets) * 100) + '%';
-    progressIndicator.style.width = value;
-
-    progressText.textContent = `Loaded ${loadedAssets} of ${totalAssets} assets`;
+  for (let i = 0; i < assets.length; i += batchSize) {
+    const batch = assets.slice(i, i + batchSize);
+    await loadBatch(batch);
   }
+}
 
-  async function load(url) {
-    return new Promise((resolve, reject) => {
-        const req = new XMLHttpRequest();
-
-        req.onload = function() {
-            if (req.status >= 200 && req.status < 300) {
-                resolve(req.response);
-            } else {
-                reject(new Error(`Failed to load: ${req.status} ${req.statusText}`));
-            }
-        };
+function reportProgress() {
```

Comment:
This is out of the scope for the refactor but I wonder if we should write some tests for this loader to just verify it works as intended

Reply:
cc: @alestiago @tomarra

Reply:
Yes, the initial effort didn't have any tests. I would very much like to have some tests on the template output that verify the loading behavior is as expected; although it might be a bit complex to assess such.